fusesoc/coremanager.py

```diff
@@ -161,6 +171,17 @@ def eq_vln(this, that):
         cores = [x["core"] for x in self._cores.values()]
         conflict_map = self._get_conflict_map()
 
+        # Enumerate cores in lock file, if any
+        lockfile_virtuals = {}
+        lockfile_cores = []
+        if isinstance(self._lockfile, dict):
```

Comment:
Can self._lockfile be anything else than a dict (if set) or None (if not set)? If so, suggest changing this to just `if self._lockfile`

Reply:
No, will use better comparison.